File: CS50/Week_6-Python/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Main FX
def main():

    letter = count_letters(text)
    word = count_words(text)
    sentence = count_sentences(text)
    index = calc_index(letter, word, sentence)

    if index >= 1 and index <= 16:
        print("Grade ", index)
    elif index > 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")

# Count number of letters present
def count_letters(string):
    letter_count = 0
    for i in string:
        # Count number of letter present
        if (i.isalpha()) == True:
            letter_count += 1
    return letter_count

# Count number of words present
def count_words(string):
    word_count = 0
    for i in string:
        # count number of spaces, allowing the counting of numbers of words
        if (i.isspace()) == True:
            word_count += 1
    return word_count + 1

# Count number of sentences present
def count_sentences(string):
    sentence_count = 0
    for i in string:
        # count punctuation to count number os sentences
        if i == '?' or i == '.' or i == '!':
            sentence_count += 1
    return sentence_count

# Calculating readability index
def calc_index(letter, word, sentence):
    avg_letter = letter * 100 / word
    avg_sentence = sentence * 100 / word
    read_index = (0.0588 * avg_letter) - (0.296 * avg_sentence) - 15.8
    return round(read_index)

# Call main FX
main()

File: CS50/Week_2-Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Declare prototype
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //TODO: Takes a text and determines its reading level.
    /*
    Example: index = 0.0588 * L - 0.296 * S - 15.8
    L is the average number of letters per 100 words in the text,
    S is the average number of sentences per 100 words in the text.
    */

    //User input
    string text = get_string("Text: ");

    //Printing number of letters
    float print_letter_count = count_letters(text);
    printf("Number of letters: %f\n", print_letter_count);

    //Printing number of words
    float print_word_count = count_words(text);
    printf("Number of words: %f\n", print_word_count);

    //Printing number of sentences
    float print_sentence_count = count_sentences(text);
    printf("Number of sentences: %f\n", print_sentence_count);

    //Compute and print grade level
    float L = 100 * (print_letter_count / print_word_count);
    float S = 100 * (print_sentence_count / print_word_count);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int round_index = round(index);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", round_index);
    }
}

int count_letters(string text)
{
    //Counting letters
    int letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //use an array to count number of characters in 'letter_count', including spaces
        char letter_count = text[i];
        //increment 'letters' if alphabet is found in 'letter_count'
        if (isalpha(letter_count))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    //Counting spaces
    int spaces = 0;
    int words = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //use an array to count number of paces in 'text'
        char space_count = text[i];
        //increment 'spaces' if  is found in 'space_count'
        //word = number of spaces plus 1
        if (isspace(space_count))
        {
            spaces++;
            words = 1 + spaces;
        }
    }
    return words;
}

int count_sentences(string text)
{
    //Counting punctuation
    int punctuation = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        //use an array to count punctuation of paces in 'text'
        char punct_count = text[i];
        //increment 'spaces' if  is found in 'punct_count'
        if (punct_count == '.' || punct_count == '?' || punct_count == '!')
        {
            punctuation++;
        }
    }
    return punctuation;
}